namui-prebuilt/src/dropdown/mod.rs

```diff
@@ -38,55 +38,53 @@ where
     pub on_select_item: TOnSelectItem,
 }
 
-impl Dropdown {
-    pub fn new() -> Self {
-        Self {
-            id: namui::nanoid(),
-            is_opened: false,
-            list_view: list_view::ListView::new(),
-            mouse_over_item_id: None,
-        }
-    }
+struct InternalProps {
+    pub rect: Rect<Px>,
+    /// Only first `is_selected = true` item will be displayed.
+    pub items: Vec<Item>,
+    /// if `visible_item_count = 0`, all items will be displayed.
+    pub visible_item_count: usize,
+    pub on_select_item: Arc<dyn Fn(String)>,
+}
 
-    pub fn update(&mut self, event: &dyn std::any::Any) {
-        if let Some(event) = event.downcast_ref::<InternalEvent>() {
-            match event {
-                InternalEvent::ToggleDropdown => {
-                    self.is_opened = !self.is_opened;
-                    self.mouse_over_item_id = None;
-                }
-                InternalEvent::MoveOverItem { item_id } => {
-                    namui::log!("MoveOverItem {}", item_id);
-                    self.mouse_over_item_id = Some(item_id.clone());
-                }
-                InternalEvent::CloseDropdown => {
-                    self.is_opened = false;
-                    self.mouse_over_item_id = None;
-                }
-            }
-        }
+pub fn render<TItems, TOnSelectItem>(props: Props<TItems, TOnSelectItem>) -> RenderingTree
+where
+    TItems: IntoIterator<Item = Item>,
+    TOnSelectItem: Fn(String) + 'static,
+{
+    let internal_props = InternalProps {
+        rect: props.rect,
+        items: props.items.into_iter().collect(),
+        visible_item_count: props.visible_item_count,
+        on_select_item: Arc::new(props.on_select_item),
+    };
+    react::<Dropdown, _, _>(
+        || {
+            Box::new(Dropdown {
+                id: namui::random_id(),
+                is_opened: false,
+                list_view: list_view::ListView::new(),
+                mouse_over_item_id: None,
+            })
+        },
+        internal_props,
+    )
```

Comment:
Dropdown become reacted. it's not stable or comfortable to make some entity as reacted. I know it and will make improvement.